plugin-core/src/main/java/com/salesforce/bazel/eclipse/classpath/BazelClasspathContainer.java

```diff
@@ -98,14 +98,20 @@
     
     private ImplicitDependencyHelper implicitDependencyHelper = new ImplicitDependencyHelper();
     
+    public BazelClasspathContainer(IProject eclipseProject)
+            throws IOException, InterruptedException, BackingStoreException, JavaModelException,
+            BazelCommandLineToolConfigurationException {
+        this(eclipseProject, BazelPluginActivator.getResourceHelper());
+    }
     
-    public BazelClasspathContainer(IProject eclipseProject, IJavaProject eclipseJavaProject)
+    BazelClasspathContainer(IProject eclipseProject, ResourceHelper resourceHelper)
             throws IOException, InterruptedException, BackingStoreException, JavaModelException,
             BazelCommandLineToolConfigurationException {
         this.eclipseProject = eclipseProject;
-        this.eclipseJavaProject = eclipseJavaProject;
         this.eclipseProjectName = eclipseProject.getName();
         this.eclipseProjectPath = eclipseProject.getLocation();
+        this.eclipseProjectIsRoot = resourceHelper.isBazelRootProject(eclipseProject);
```

Comment:
Instead of checking multiple times in logic below whether this project is the root project, check here once.